### ISSUE TITLE
keyNested failing to validate existence of key

### ISSUE THREAD

**Issue by PatrickSJ** [Respect/Validation#886]
According to the docs I can validate whether a key exists with KeyNested, but if I lack a key (see below), I get a PHP Fatal Error.   I was expecting an exception, not an error.
 

```
   $array = array('foo' => array('bad_key' => ''));
    $v = Respect\Validation\Validator::keyNested('foo')->keyNested('foo.bar');

    try {
        $v->assert($array);
        echo "Did not catch foo.bar being missing" . PHP_EOL;
    }
    catch (Respect\Validation\NestedValidationException $e) {
        $e->getMessages();
    }
```

```
PHP Fatal error:  Uncaught These rules must pass for { "foo": { "bad_key": "" } }
  thrown in /Users/patrick/Sites/SolutionModule/vendor/respect/validation/library/Validator.php on line 248

Fatal error: Uncaught These rules must pass for { "foo": { "bad_key": "" } }
  thrown in /Users/patrick/Sites/SolutionModule/vendor/respect/validation/library/Validator.php on line 248

Respect\Validation\Exceptions\AllOfException: These rules must pass for { "foo": { "bad_key": "" } } in /Users/patrick/Sites/SolutionModule/vendor/respect/validation/library/Validator.php on line 248

```

**Comment by PatrickSJ** [comment]
Don't know where my brain was.  Need to catch Exception, not Respect\Validation\NestedValidationException.  No issue.
